bioblend/galaxy/jobs/__init__.py

```diff
@@ -129,3 +131,36 @@ def get_metrics(self, job_id):
         """
         url = self._make_url(module_id=job_id) + '/metrics'
         return self._get(url=url)
+
+    def report_error(self, job_id, dataset_id, message, email=None):
+        """
+        Report an error for a given job and dataset.
+
+        :type job_id: str
+        :param job_id: job ID
+
+        :type dataset_id: str
+        :param dataset_id: Dataset ID
+
+        :type message: str
+        :param message: Error message
+
+        :type email: str
+        :param email: Email to submit error report to
+
+        :rtype: dict
+        :return: dict containing job error reply
+
+        This method reports an error for a job and dataset to Galaxy.
+        """
+        payload = {
+            "message": message,
+            "dataset_id": dataset_id
+        }
+        if email is not None:
+            payload["email"] = email
+
+        url = self._make_url(module_id=job_id) + '/error'
+        return self._post(url=url, payload=urlencode(payload, quote_via=quote_plus),
+                          headers_override={'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'}
```

Comment:
I didn't have to time to check, but is it actually necessary to urlencode the payload and add the `headers_override` ? The payload is already serialised in JSON by `make_post_request()`.

Reply:
From my testing, this particular endpoint didn't accept 'application/json' as Content-Type.

Reply:
That is a problem in the Galaxy API, will be fixed by https://github.com/galaxyproject/galaxy/pull/9753